### DISCUSSION TITLE
ModuleNotFoundError even after upgrading and force-reinstall

### DISCUSSION THREAD

**Discussion by alecristia** [LAAC-LSCP/ChildProject#265] [Q&A] [answered]
When trying to re-calculate reliability for EL1000, I get: `ModuleNotFoundError: No module named 'pyannote.metrics'`

Following [installation instructions](https://childproject.readthedocs.io/en/latest/install.html), I did

pip3 uninstall ChildProject and then pip3 install ChildProject --upgrade.

as well as

pip3 install git+https://github.com/LAAC-LSCP/ChildProject.git --force-reinstall

But I can't shake that error. 

The output of the commands above definitely includes pyannote:

> Collecting pyannote.database>=4.1
  Downloading pyannote.database-4.1.1-py3-none-any.whl (41 kB)
     |████████████████████████████████| 41 kB 31 kB/s 
Collecting pyannote.core>=4.1
  Downloading pyannote.core-4.1-py3-none-any.whl (56 kB)

and later

> Installing collected packages: six, pycparser, numpy, urllib3, scipy, pytz, python-dateutil, pyparsing, pillow, kiwisolver, idna, cycler, click, chardet, cffi, certifi, zipp, wrapt, typing-extensions, typer, sortedcontainers, simplejson, shellingham, setuptools, requests, qdldl, pandas, matplotlib, llvmlite, jeepney, dataclasses, cryptography, colorama, threadpoolctl, SecretStorage, scs, requests-toolbelt, PyYAML, pynacl, pyjwt, pyannote.core, packaging, osqp, numba, joblib, importlib-metadata, future, ecos, deprecated, appdirs, whoosh, tqdm, TextGrid, soundfile, scikit-learn, resampy, regex, redo, python-magic, python-gitlab, pympi-ling, PyGithub, pyannote.database, pooch, patool, msgpack, MarkupSafe, keyrings.alt, keyring, jsmin, iso8601, humanize, fasteners, decorator, cvxpy, cvxopt, boto, audioread, annexremote, sox, sklearn, pysoundfile, pylangacq, pygamma-agreement, pydub, panoptes-client, nltk, lxml, librosa, jinja2, importlib-resources, datalad, ChildProject

and also

> Successfully installed ChildProject-0.0.1 MarkupSafe-2.0.1 PyGithub-1.55 PyYAML-5.4.1 SecretStorage-3.3.1 TextGrid-1.5 annexremote-1.5.0 appdirs-1.4.4 audioread-2.1.9 boto-2.49.0 certifi-2021.5.30 cffi-1.14.6 chardet-3.0.4 click-7.1.2 colorama-0.4.4 cryptography-3.4.7 cvxopt-1.2.6 cvxpy-1.1.15 cycler-0.10.0 dataclasses-0.8 datalad-0.14.7 decorator-5.0.9 deprecated-1.2.12 ecos-2.0.7.post1 fasteners-0.16.3 future-0.18.2 humanize-3.11.0 idna-2.10 importlib-metadata-4.6.4 importlib-resources-5.2.2 iso8601-0.1.16 jeepney-0.7.1 jinja2-3.0.1 joblib-1.0.1 jsmin-2.2.2 keyring-23.1.0 keyrings.alt-4.1.0 kiwisolver-1.3.1 librosa-0.8.1 llvmlite-0.36.0 lxml-4.6.3 matplotlib-3.3.4 msgpack-1.0.2 nltk-3.6.2 numba-0.53.1 numpy-1.19.5 osqp-0.6.2.post0 packaging-21.0 pandas-1.1.5 panoptes-client-1.4.0 patool-1.12 pillow-8.3.1 pooch-1.4.0 pyannote.core-4.1 pyannote.database-4.1.1 pycparser-2.20 pydub-0.25.1 pygamma-agreement-0.4.0 pyjwt-2.1.0 pylangacq-0.15.0 pympi-ling-1.70.2 pynacl-1.4.0 pyparsing-2.4.7 pysoundfile-0.9.0.post1 python-dateutil-2.8.2 python-gitlab-2.10.0 python-magic-0.4.24 pytz-2021.1 qdldl-0.1.5.post0 redo-2.0.4 regex-2021.8.3 requests-2.24.0 requests-toolbelt-0.9.1 resampy-0.2.2 scikit-learn-0.24.2 scipy-1.5.4 scs-2.1.4 setuptools-57.4.0 shellingham-1.4.0 simplejson-3.17.3 six-1.16.0 sklearn-0.0 sortedcontainers-2.4.0 soundfile-0.10.3.post1 sox-1.4.1 threadpoolctl-2.2.0 tqdm-4.62.1 typer-0.3.2 typing-extensions-3.10.0.0 urllib3-1.25.11 whoosh-2.7.4 wrapt-1.12.1 zipp-3.5.0

I'm working in oberon in my ChildProjectVenv.

Just in case, I'm updating my pip and force-reinstalling again...

**Comment by lucasgautheron** [comment, marked as answer]
Try `pip install pyannote.metrics` ? not sure why it won't install by itself
EDIT: oh I get it; you need python >= 3.7 (pyannote metrics may have issues with older versions of python)

But I suggest you use python 3.8 at least.
Maybe you should delete your venv and create it again:

```bash
# delete the environment
deactivate
rm -rf ~/ChildProjectVenv
conda activate base

# make sure your version of python is recent enough (3.8+)
python3 --version

# recreate the environment
python3 -m venv ~/ChildProjectVenv
source ~/ChildProjectVenv/bin/activate
pip install --upgrade pip
pip3 install ChildProject
```

**Reply by alecristia** [reply]
notice in particular that my copy-paste on what had been installed was skipping pyannote.metrics. Here's what it should look like: 

> Successfully installed ChildProject-0.0.1 MarkupSafe-2.0.1 PyGithub-1.55 PyYAML-5.4.1 SecretStorage-3.3.1 TextGrid-1.5 annexremote-1.5.0 appdirs-1.4.4 audioread-2.1.9 boto-2.49.0 certifi-2021.5.30 cffi-1.14.6 chardet-3.0.4 click-7.1.2 colorama-0.4.4 cryptography-3.4.7 cvxopt-1.2.6 cvxpy-1.1.15 cycler-0.10.0 datalad-0.14.7 decorator-5.0.9 deprecated-1.2.12 docopt-0.6.2 ecos-2.0.7.post1 fasteners-0.16.3 future-0.18.2 humanize-3.11.0 idna-2.10 importlib-metadata-4.6.4 importlib-resources-5.2.2 iso8601-0.1.16 jeepney-0.7.1 jinja2-3.0.1 joblib-1.0.1 jsmin-2.2.2 keyring-23.1.0 keyrings.alt-4.1.0 kiwisolver-1.3.1 librosa-0.8.1 llvmlite-0.36.0 lxml-4.6.3 matplotlib-3.4.3 mpmath-1.2.1 msgpack-1.0.2 nltk-3.6.2 numba-0.53.1 numpy-1.21.2 osqp-0.6.2.post0 packaging-21.0 pandas-1.1.5 panoptes-client-1.4.0 patool-1.12 pillow-8.3.1 pooch-1.4.0 pyannote.core-4.1 pyannote.database-4.1.1 pyannote.metrics-3.0.1 pycparser-2.20 pydub-0.25.1 pygamma-agreement-0.4.0 pyjwt-2.1.0 pylangacq-0.15.0 pympi-ling-1.70.2 pynacl-1.4.0 pyparsing-2.4.7 pysoundfile-0.9.0.post1 python-dateutil-2.8.2 python-gitlab-2.10.0 python-magic-0.4.24 pytz-2021.1 qdldl-0.1.5.post0 redo-2.0.4 regex-2021.8.3 requests-2.24.0 requests-toolbelt-0.9.1 resampy-0.2.2 scikit-learn-0.24.2 scipy-1.7.1 scs-2.1.4 shellingham-1.4.0 simplejson-3.17.4 six-1.16.0 sklearn-0.0 sortedcontainers-2.4.0 soundfile-0.10.3.post1 sox-1.4.1 sympy-1.8 tabulate-0.8.9 threadpoolctl-2.2.0 tqdm-4.62.1 typer-0.3.2 typing-extensions-3.10.0.0 urllib3-1.25.11 whoosh-2.7.4 wrapt-1.12.1 zipp-3.5.0
